src/SFA.DAS.Funding.SystemAcceptanceTests/Helpers/AzureServiceBusClient.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Azure.Identity;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+
+namespace SFA.DAS.Funding.SystemAcceptanceTests.Helpers
+{
+    internal class AzureServiceBusClient
+    {
+        private readonly ServiceBusAdministrationClient _administrationClient;
+        private static string TopicSubscriptionFilterName => "AllEvents";
+
+        public AzureServiceBusClient(string azureServiceBusNamespace)
+        {
+            _administrationClient = new ServiceBusAdministrationClient(azureServiceBusNamespace, new DefaultAzureCredential());
+        }
+
+        internal async Task CreateSubscriptionWithFiltersAsync(string subscriptionName, string topicName,
+            string destinationQueueName, List<string>? filterEventTypes)
+        {
+            await CreateSubscriptionAsync(subscriptionName, topicName, destinationQueueName);
+
+            if (filterEventTypes is { Count: > 0 })
+            {
+                await DeleteDefaultRuleAsync(subscriptionName, topicName);
+                await CreateNewSqlFilter(subscriptionName, topicName, filterEventTypes);
+            }
+        }
+
+        private async Task CreateNewSqlFilter(string subscriptionName, string topicName, List<string> filterEventTypes)
+        {
+            try
+            {
+                var sqlExpression = "[NServiceBus.EnclosedMessageTypes] LIKE '%" +
+                                    string.Join("%' OR [NServiceBus.EnclosedMessageTypes] LIKE '%",
+                                        filterEventTypes) + "%'";
```

Comment:
saw your note on the main conversation page about this, only questions I have are:
 - what is the purpose of the first part of this string before the OR? [NServiceBus.EnclosedMessageTypes] LIKE '%%'
 - could we either parameterise this and/or pass the list of types all the way down to here and have this build the sql expression from that list? just reduces any risk of someone calling the public method on this class with some garbage string later on

Reply:
I now realise this will require parsing of event types from config, but possibly worth it? Either that or hardcode them in a constants class (it's not like we will be subscribing to and handling new event types without a code change)

Reply:
you could even reflect over types that inherit from MultipleEndpointSafeEventHandler and use the <T> generic types to build the list